File: desktop/src/com/inkostilation/pong/desktop/PongMain.java
package com.inkostilation.pong.desktop;

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Game;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.glutils.ShapeRenderer;
import com.inkostilation.pong.desktop.controls.InputSystem;
import com.inkostilation.pong.desktop.display.LobbyScreen;
import com.inkostilation.pong.desktop.display.PongScreen;
import com.inkostilation.pong.desktop.network.Network;
import com.inkostilation.pong.desktop.network.ServerThread;
import com.inkostilation.pong.engine.Ball;
import com.inkostilation.pong.engine.Field;
import com.inkostilation.pong.engine.IEngine;
import com.inkostilation.pong.engine.Paddle;

public class PongMain extends Game {

	@Override
	public void create () {
		setScreen(new LobbyScreen());
		ServerThread.getInstance().start();
	}

	@Override
	public void render () {
		super.render();
	}
	
	@Override
	public void dispose () {
		super.dispose();
		ServerThread.getInstance().stopServer();
		while (!ServerThread.getInstance().isFinished()) {
			try {
				Thread.sleep(100);
			} catch (InterruptedException e) {
				e.printStackTrace();
			}
		}
	}
}

File: desktop/src/com/inkostilation/pong/desktop/PongMain.kt
package com.inkostilation.pong.desktop

import com.badlogic.gdx.Game
import com.inkostilation.pong.client.NetworkConnector
import com.inkostilation.pong.commands.AbstractRequestCommand
import com.inkostilation.pong.commands.request.CounterCommand
import com.inkostilation.pong.commands.request.InitCommand
import com.inkostilation.pong.commands.request.RequestMessageCommand
import com.inkostilation.pong.server.engine.AbstractEngine
import com.inkostilation.pong.server.engine.IEngine

class PongMain : Game() {

    private lateinit var networkConnector: NetworkConnector<Game>

    override fun create() {
        networkConnector = NetworkConnector(this, "localhost", 8080)
        networkConnector.start()
        val command = InitCommand()
        networkConnector.send(command)

        val counter = CounterCommand()
        while (true) {
            Thread.sleep(500)
            networkConnector.send(counter)
        }
    }

    override fun render() {
        super.render()
    }

    override fun dispose() {
        super.dispose()
        networkConnector.stop()
    }
}